bundle.js

```diff
@@ -126,12 +126,40 @@ const addRequest = (event, bundle, convertedBundle) => {
   const params = _.get(bundle, 'request.params', {});
   _.extend(convertedBundle.request.params, params);
 
-  let body = _.get(bundle, 'request.body');
+  const body = _.get(bundle, 'request.body');
   if (body) {
-    if (typeof body !== 'string') {
-      body = JSON.stringify(body);
+    let data = body,
+      files;
+
+    if (typeof data !== 'string') {
+      if (bundle._fileFieldKeys) {
+        // Exclude file fields from request.data
+        data = Object.keys(body)
+          .filter(k => bundle._fileFieldKeys.indexOf(k) === -1)
+          .reduce((result, k) => {
+            result[k] = body[k];
+            return result;
+          }, {});
+
+        files = Object.keys(body)
+          .filter(k => bundle._fileFieldKeys.indexOf(k) >= 0)
+          .reduce((result, k) => {
+            // TODO: Any way we can tell filename and mimetype from a hydrate URL?
+            const filename = 'filename';
+            const mimetype = 'application/octet-stream';
+            result[k] = [filename, body[k], mimetype];
+            return result;
+          }, {});
+      }
+
+      data = JSON.stringify(data);
+    }
+
+    convertedBundle.request.data = data;
+    if (!_.isEmpty(files)) {
+      convertedBundle.request.files = files;
+      delete convertedBundle.request.headers['Content-Type'];
```

Comment:
Is this just deleting the default header we would have set? Asking because `zapier/zapier` had some logic to try and [preserve the content type of the form data being sent](https://github.com/zapier/zapier/blob/3d9260a67fdb235b68ca8f7c56e068f7a388aad4/developer/api_base.py#L1759-L1768) in the multipart request.

Reply:
This is to remove the `application/json` header that `bundleConverter` set earlier [here](https://github.com/zapier/zapier-platform-legacy-scripting-runner/blob/0306fc81642b2dedb469216f9f7b615a84940196/bundle.js#L162). Thanks for pointing it out!